packages/at_secondary_server/lib/src/connection/outbound/outbound_message_listener.dart

```diff
@@ -16,17 +16,21 @@ class OutboundMessageListener {
 
   /// Listens to the underlying connection's socket if the connection is created.
   /// @throws [AtConnectException] if the connection is not yet created
-  void listen() {
-    var connection = outboundClient.outboundConnection!;
-    connection.getSocket().listen(_messageHandler,
+  void listen() async {
+    outboundClient.outboundConnection?.getSocket().listen(_messageHandler,
         onDone: _finishedHandler, onError: _errorHandler);
     _queue = Queue();
-    connection.getMetaData().isListening = true;
+    outboundClient.outboundConnection?.getMetaData().isListening = true;
   }
 
   /// Handles responses from the remote secondary, adds to [_queue] for processing in [read] method
   /// Throws a [BufferOverFlowException] if buffer is unable to hold incoming data
   Future<void> _messageHandler(data) async {
+    //ignore the data if connection is closed or stale
+    if (outboundClient.outboundConnection!.getMetaData().isStale || outboundClient.outboundConnection!.getMetaData().isClosed){
```

Comment:
code formatting missing

Reply:
formatted code pushed in the latest commit